internal/util/importutil/json_parser.go

```diff
@@ -0,0 +1,238 @@
+package importutil
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/proto/schemapb"
+)
+
+const (
+	// root field of row-based json format
+	RowRootNode = "rows"
+	// initial size of a buffer
+	BufferSize = 1024
+)
+
+type JSONParser struct {
+	ctx     context.Context  // for canceling parse process
+	bufSize int64            // max rows in a buffer
+	fields  map[string]int64 // fields need to be parsed
+}
+
+// newImportManager helper function to create a importManager
+func NewJSONParser(ctx context.Context, collectionSchema *schemapb.CollectionSchema) *JSONParser {
```

Comment:
Write a jsonparser from scratch is a tricky,
It would be much better if we import json parser with iterator such as https://github.com/buger/jsonparser

Reply:
The JSONParser is only a wrapper of json.Decoder. If the **github.com/buger/jsonparser** has the ability to read json file by streaming, then we can replace the logic in the JSONParser. From the **github.com/buger/jsonparser** source code I didn't see the example how it can streaming parse json file, so keep this as an future option.